File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def is_same(s1, s2):
    return (s1 == "(" and s2 == ")") or (s1 == "[" and s2 == "]") or s1 == "{" and s2 == "}"

def is_right(s):
    
    stack = []
    
    for _ in range(len(s)):
        if stack and s and is_same(s[-1], stack[-1]):
            s = s[:-1]
            stack.pop()
        else:
            char = s[-1]
            s = s[:-1]
            stack.append(char)
            
    return len(stack) == 0

def solution(s):
    
    ss = s + s
    
    answer = 0
    
    for i in range(len(s)):
        if is_right(ss[i:i+len(s)]):
            answer += 1
    
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.swift
import Foundation

func isSame(_ s1: Character, _ s2: Character) -> Bool {
    return 
        s1 == "(" && s2 == ")"
        || s1 == "[" && s2 == "]"
        || s1 == "{" && s2 == "}"
}

func isRight(_ arr: [Character]) -> Bool {
    
    var arr = arr
    var stack: [Character] = []
    
    while !arr.isEmpty {
        if !stack.isEmpty && isSame(arr.last!, stack.last!) {
            stack.removeLast()  
            arr.removeLast()
        } else {
            stack.append(arr.popLast()!)
        }
    }
    
    
    
    return stack.isEmpty
}

func solution(_ s:String) -> Int {
    
    let arr = Array(s) + Array(s)
    var answer = 0
    
    for i in 0..<s.count {
        let isRight = isRight(Array(arr[i..<i+s.count]))
        print(isRight)
        answer += isRight ? 1 : 0
    }
    
    return answer
}